### PR TITLE
Add new endpoint to resolve block hash by slot

### PR DESCRIPTION
**Pull request by AhmedIbrahim336**
### Purpose
Resolve the latest block created at a slot equal to or lower than a given slot.

### Request 

The field slots are required to be a non-empty array. Each entry is a tuple of two numbers: epoch and slot.

  ```ts
  type SlotNo = number;
  type EpochNo = number;
  type Request = {
    slots: Array<[EpochNo, SlotNo]>
  };
  ```

  ### Response 

  ```ts
  type Slot = [EpochNo, SlotNo];
  type Response = {
    blockHashes: {
      [key: Slot]: string | null
    }
  }
  ```

---- 
For detailed overview you can check [YB-113](https://emurgo.atlassian.net/browse/YB-113)

### CODE REVIEW

File: config/default.ts [hunk 1]
```diff
@@ -1,3 +1,7 @@
+import dotenv from "dotenv";
+
+dotenv.config({ path: "./.env" });
```
Comment (AhmedIbrahim336) [inline config/default.ts:3]: This is not relevant to this PR. I just included it to load the env variables from the `.env` file so we don't have to add them every time.

File: README.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ npm install
 npm run dev
 ```
 
-The server will then run at http://localhost:8082. You can query using curl (ex: `curl http://localhost:8082/bestblock`)
+The server will then run at http://localhost:8082. You can query using curl (ex: `curl http://localhost:8082/v2/bestblock`)
```
Comment (AhmedIbrahim336) [inline README.md:39]: Unrelatedly to this PR. I fixed it as the old example doesn't work.

File: config/default.ts (hunk 1 above)
Comment (vsubhuman) [reply config/default.ts:3]: nice

File: src/services/getBlockHash.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import { Request, Response } from "express";
+import { Pool } from "pg";
+
+const query = `
+    SELECT encode(hash, 'hex') as hash, epoch_no, slot_no
+    FROM   block
+    WHERE  epoch_no <= $1 AND slot_no <= $2
```
Comment (vsubhuman) [inline src/services/getBlockHash.ts:7]: @AhmedIbrahim336 , note that slot `[1, 20]` is lower than slot `[2, 10]`, but your query will not select it.

**Review by vsubhuman — CHANGES_REQUESTED** [review]

**Review by vsubhuman — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -11,7 +11,8 @@
     "dev": "tsc-watch --onSuccess \"node ./dist/index.js\"",
     "eslint": "eslint src tests",
     "test": "mocha -r ts-node/register tests/**/*.test.ts --slow 0",
-    "testtxhist": "mocha -r ts-node/register tests/txHistory.test.ts",
+    "test:file": "mocha -r ts-node/register ",
+    "testtxhist": "npm run test:file -- tests/txHistory.test.ts",
```
Comment (AhmedIbrahim336) [inline package.json:15]: Unrelated: add new script `test:file` to run a specific test file(s) which can be passed as a parameter using npm. e.g. `npm run test:file -- test/txHistory.test.ts`

**Comment by AhmedIbrahim336** [comment]
This change has been impelmented in the [neo4j](https://github.com/Emurgo/yoroi-graphql-migration-backend/tree/neo4j) branch

**Pull request closed without merging**